lib/hacluster.pm

```diff
@@ -979,6 +980,44 @@ sub activate_ntp {
     systemctl "enable --now $ntp_service.service";
 }
 
+=head2 script_output_retry_check
+
+  script_output_retry_check(cmd=>$cmd, regex_string=>$regex_sring, [retry=>$retry, sleep=>$sleep, ignore_failure=>$ignore_failure]);
+
+Executes command via 'script_output' subroutine and makes a sanity check against a regular expression. Command output is returned
+after success, otherwise the command is retried defined number of times. Test dies after last unsuccessfull retry.
+
+C<$cmd> command being executed.
+C<$regex_string> regular expression to check output against.
+C<$retry> number of retries. Defaults to C<5>.
+C<$sleep> sleep time between retries. Defaults to C<10s>.
+C<$ignore_failure> do not kill the test upon failure.
+
+  Example: script_output_retry_check(cmd=>'hostname', regex_string=>'^node01$', retry=>'100', sleep=>'60', ignore_failure=>'1');
+
+=cut
+
+sub script_output_retry_check {
+    my %args = @_;
+    my $cmd = $args{cmd} // die('No command specified.');
+    my $regex = $args{regex_string} // die('Regex input missing');
+    my $retry = $args{retry} // 5;
+    my $sleep = $args{sleep} // 10;
+    my $ignore_failure = $args{ignore_failure} // "0";
+    my $result;
+
+    foreach (1 .. $retry) {
+        $result = script_output($cmd);
```

Comment:
You could also do:
```
$result = script_output($cmd, %args)
```
And the extra args you receive in this function can be passed to the inner function.